driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java

```diff
@@ -289,7 +291,13 @@ public void refreshSchema(String keyspace, String table) throws InterruptedExcep
             // At startup, when we add the initial nodes, this will be null, which is ok
             if (c == null)
                 return;
-            refreshSchema(c, keyspace, table, cluster, false);
+            refreshSchema(c, keyspace, table, cluster);
+            // If the table is null, we either rebuild all from scratch or have an updated keyspace. In both cases, rebuild the token map
+            // since some replication on some keyspace may have changed
+            if (table == null) {
+                cluster.submitNodeListRefresh();
+                cluster.nodeListRefreshRequestDebouncer.deliverEvents().get();
```

Comment:
This can get weird since we could be inside a SchemaRefreshRequestDeliveryCallback which is already executing in Cluster.executor, this will submit another task to Cluster.executor.  This gets even weirder if the schemaRefresh was triggered by schemaRefreshRequestDebouncer.deliverEvents in Cluster.maybeRefreshSchemaAndSignal.  This leaves us with a situation where we have:

    - A response submitting SchemaRefreshDeiveryCallback in Cluster.executor via maybeRefreshSchemaAndSingal
        - SchemaRefreshDeliveryCallback submitting Runnable to Cluster.executor:
            - Runnable calling controlConnection.refreshSchema which submits a nodeListRefresh
                -  NodeListRefreshDeliveryCallback submitting a Runnable to refreshNodeListAndTokenMap in Cluster.Executor.

If Cluster.executor only has 2 threads (dual core or less system or configured that way), wouldn't starve the executor?